### PR TITLE
Covid19 Referral Form

### PR DESCRIPTION
**Pull request by ODORA0**
https://app.asana.com/0/1133167201254915/1200440304211289

### CODE REVIEW

**Comment by slubwama** [comment]
@semujju  @Mwanje Why is it that is form doesn't have any review

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+                                                <label>Recieving Facility</label>
+                                                <obs conceptId="159395" />
+                                            </div>
+                                            <div class="col-md-3">
+                                                <label>Ward/Unit</label>
+                                                <obs conceptId="159395" />
+                                            </div>
+                                            <div class="col-md-3">
+                                                <label>Recieving Clinician</label>
+                                                <encounterProviderAndRole encounterRole="240b26f9-dd88-4172-823d-4a8bfeb7841f" required="required" />
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: How do we differentiate  between a Medical Officer. Looks to me they share the same enounterRole, which makes it difficult to separate that data if we needed to report about medical officers and referring clinician.

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+                                            </div>
+                                        </div>
+                                        <div class="row">
+                                            <div class="col-md-3">
+                                                <label>Designation</label>
+                                                <obs conceptId="159395" />
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Can this be wrapped in a grouping observation

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+                            </div>
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Can this be wrapped in a grouping observation

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+
+                            <div class="tab-pane active" id="patient-history">
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Remove "active" class among classes. this is causing the tab content  on load to be in the first tab

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+                                                    <obs id="alcohol" conceptId="166351" answerConceptId="794" answerLabel="Alcohol" style="checkbox" class="col-md-3" />
+                                                    <obs id="illicit-drug-use" conceptId="166351" answerConceptId="166349" answerLabel="Illicit Drug Use" style="checkbox" class="col-md-3" />
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+
+                            <div class="tab-pane active" id="examinations">
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Remove "active" class among classes. this is causing the tab content  on load to be in the first tab

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+                                            <div class="col-md-4">
+                                                <label>Key Systemic Exam Finding</label>
+                                                <obs conceptId="159395" style="textarea" />
+                                            </div>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+
+                            <div class="tab-pane active" id="current-management-and-further-plan">
```
Comment (slubwama) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Remove "active" class among classes. this is causing the tab content  on load to be in the first tab

**Comment by semujju** [comment]
@ODORA0 
Referral Details

- Covid Status and Covid Severity - be in the same row (use col-md-6 for each)
- Covid Severity - use drop down
- Covid Status - Answers should be in sentence case
- Reason for referral and Medical Officer - be on the same row (use col-md-6 for each)
- Reason for referral - be sentence case

Referring Facility/Recieving Facility

- Referring Facility/Recieving Facility and Ward/Unit - be in the same row (use col-md-4 for each)
- Referring Clinician/Recieving Clinician, Phone and Designation - be in the same row (use col-md-4 for each)
- Write Phone Number instead of Phone

Patient History

- Put social History after Chronic Conditions/PMH
- Add Occupation
- Chronic Medication - Pick code from Art card (medication bit that has drug,dose duration etc)
- Add Admission date to current facility
- Presenting Complaint and History - Remove

Examination

- Add General (textbox)
- Key Systemic Exam Findings (textbox)

Vital Signs Progress

- Look at Clinical Management Form - Vitals card (patient information)
- Create Two Cards, one for arrival and another on for current

Investigations

- Look at Clinical Management Form - Lab test and results section

Current Management

- Medication - Add date, present medication as it is on art card
- For oxygen,intubated cases and ventilator settings - borrow entire "supportive care tab" from Clinical Management Form

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml (hunk 1 above)
Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: I think the best way is to wrap them under grouping concepts

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml (hunk 2 above)
Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Sure

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml (hunk 3 above)
Comment (semujju) [reply omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml]: Sure

File: omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml [hunk 7]
```diff
@@ -0,0 +1,1252 @@
...
+                                                        <obs id="arrival" conceptId="159395" answerLabel="arrival" class="col-md-3" />
+                                                    </div>
+                                                    <div class="col-md-4">
+                                                        <label>Current</label>
+                                                        <obs conceptId="159395" class="col-md-3" />
+                                                    </div>
+                                                </div>
+                                            </div>
+                                            <div class="col-md-6">
+                                                <label>AVPU(Alert, to Pain, to
```
Comment (semujju) [inline omod/src/main/webapp/resources/htmlforms/Covid19-ReferralForm.xml:871]: use this concept (166410) for AVPU and present answers as drop down, remove them from brackets in the label

**Pull request merged**